core/domain/stats_services_test.py

```diff
@@ -1213,6 +1162,49 @@ def test_update_exp_issues_for_new_exp_version_with_cyclic_state_transition(
                 'state_name']['value'],
             'Renamed state')
 
+        # Test deleting of states.
+        exploration.delete_state('End')
+        exploration.version += 1
+        change_list = [exp_domain.ExplorationChange({
+            'cmd': 'delete_state',
+            'state_name': 'End',
+        })]
+        exp_versions_diff = exp_domain.ExplorationVersionsDiff(change_list)
+        stats_services.update_exp_issues_for_new_exp_version(
+            exploration, exp_versions_diff, False)
+
+        exp_issues = stats_services.get_exp_issues(
+            exploration.id, exploration.version)
+        self.assertEqual(exp_issues.exp_version, 3)
+        self.assertFalse(exp_issues.unresolved_issues[0].is_valid)
+        self.assertEqual(
+            exp_issues.unresolved_issues[0].issue_customization_args[
+                'state_names']['value'], [
+                    'Renamed state', 'End', 'Renamed state'])
+
+        playthrough_instance = stats_models.PlaythroughModel.get(
+            playthrough_id)
+        self.assertEqual(
+            playthrough_instance.issue_customization_args['state_names'][
+                'value'], ['Renamed state', 'End', 'Renamed state'])
+        self.assertEqual(
+            playthrough_instance.actions[0]['action_customization_args'][
+                'state_name']['value'],
+            'Renamed state')
+
+    def test_delete_playthroughs_multi(self):
+        playthrough_ids = [
+            stats_models.PlaythroughModel.create(
+                'exp_id1', 1, 'EarlyQuit', {}, []),
+            stats_models.PlaythroughModel.create(
+                'exp_id1', 1, 'EarlyQuit', {}, []),
+        ]
+
+        stats_services.delete_playthroughs_multi(playthrough_ids)
```

Comment:
Is this test correct? It seems to be passing in a list of models, not IDs. The function contract seems to be expecting IDs, as well.

Reply:
Yep it's correct, the `create` method returns the ID of the new model. This is the same contract used by GAE's put: https://cloud.google.com/appengine/docs/standard/python/ndb/modelclass#instance_methods.